server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -609,57 +704,45 @@ private static SnapshotsInRepo sortSnapshots(
         @Nullable String snapshotName,
         @Nullable String repoName
     ) {
-        final Predicate<SnapshotInfo> isAfter;
         switch (sortBy) {
             case START_TIME:
-                isAfter = filterByLongOffset(SnapshotInfo::startTime, Long.parseLong(after), snapshotName, repoName, order);
-                break;
+                return filterByLongOffset(SnapshotInfo::startTime, Long.parseLong(after), snapshotName, repoName, order);
             case NAME:
                 if (snapshotName == null) {
-                    assert repoName == null : "no snapshot name given but saw repo name [" + repoName + "]";
-                    isAfter = order == SortOrder.ASC
-                        ? snapshotInfo -> after.compareTo(snapshotInfo.snapshotId().getName()) <= 0
-                        : snapshotInfo -> after.compareTo(snapshotInfo.snapshotId().getName()) >= 0;
+                    // already covered by preflight filtering
+                    return null;
                 } else {
-                    isAfter = order == SortOrder.ASC
+                    // TODO: cover via pre-flight predicate
```

Comment:
Would it be too complex/noisy to address in this PR too?

Reply:
Annoyingly enough yes because of https://github.com/elastic/elasticsearch/pull/78032#discussion_r715637376